### PR TITLE
GAWB-1258 Split Rawls into rawlsModel and rawlsCore and build separately

### PR DESCRIPTION
**Pull request by helgridly**
Part one of a slow crawl towards using the model classes in Orchestration. This reorganizes the directory structure to `core/` and `model/`, where the latter is ultimately built into a separate JAR (that we intend to publish to Artifactory in GAWB-1259).

It also breaks out our build.sbt into something that approximates best practices for sbt multi-project builds.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by helgridly** [comment]
@kshakir @geoffjentry Can I tap one of my fellow SBT wizards for a review of this?

File: build.sbt [hunk 1]
```diff
@@ -86,72 +13,18 @@ javaOptions in Revolver.reStart := sys.env("JAVA_OPTS").split(" ")
 
 net.virtualvoid.sbt.graph.Plugin.graphSettings
 
-def isIntegrationTest(name: String) = name contains "integrationtest"
+lazy val rawlsModel: Project = (project in file("model"))
```
Comment (kshakir) [inline build.sbt]: The other two `Project`s below don't use an annotated return type, nor infix notation.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "io.spray" %% "spray-http" % sprayV,
+    "joda-time" % "joda-time" % "2.9.4",
+    "org.joda" % "joda-convert" % "1.8",
+    "com.typesafe.scala-logging" %% "scala-logging" % "3.1.0",
+    "org.scalatest" %% "scalatest" % "2.2.4" % "test"
+  )
+
+  val rawlsDependencies = modelDependencies ++ Seq(
+    "com.typesafe" % "config" % "1.3.0",
+    ("com.gettyimages" %% "spray-swagger" % "0.5.0").exclude("com.typesafe.scala-logging", "scala-logging-slf4j_2.11").exclude("com.typesafe.scala-logging", "scala-logging-api_2.11").exclude("com.google.guava", "guava"),
```
Comment (kshakir) [inline project/Dependencies.scala]: Perhaps convert to the more conventional infix DSL:

``` scala
"com.gettyimages" %% "spray-swagger" % "0.5.0"
  exclude("com.typesafe.scala-logging", "scala-logging-slf4j_2.11")
  exclude("com.typesafe.scala-logging", "scala-logging-api_2.11")
  exclude("com.google.guava", "guava"),
```

Same goes for all other calls to the `%` operator that end up wrapped in parenthesis below.

File: project/Settings.scala [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+object Settings {
+
+  val artifactory = "https://artifactory.broadinstitute.org/artifactory/"
+
+  val commonResolvers = List(
+    "artifactory-releases" at artifactory + "libs-release",
+    "artifactory-snapshots" at artifactory + "libs-snapshot"
+  )
+
+  val buildSettings: Seq[_root_.sbt.Def.Setting[_]] = Defaults.defaultSettings ++ Seq(
```
Comment (kshakir) [inline project/Settings.scala]: Unless there's a conflict, `_root_.sbt.Def.` isn't needed here nor below on line 38.

Also, I'm not sure what `Defaults.defaultSettings` is nor what it does, but its use is deprecated by sbt.

``` scala
@deprecated("Default settings split into coreDefaultSettings and IvyModule/JvmModule plugins.", "0.13.2")
lazy val defaultSettings: Seq[Setting[_]] = projectBaseSettings ++ defaultConfigs
```

File: project/Settings.scala [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    buildSettings ++ assemblySettings ++ testSettings ++ List(
+    organization  := "org.broadinstitute",
+    scalaVersion  := "2.11.7",
+    resolvers ++= commonResolvers,
+    scalacOptions ++= compilerSettings
+  )
+
+  val modelSettings = List(
+    name := "rawls-model",
+    version := "0.1", //FIXME: do we ever increment this? Should we care?
```
Comment (kshakir) [inline project/Settings.scala:48]: IMHO, all three sub-projects should be using a common version constant, to be defined somewhere above.

**Review by kshakir — COMMENTED** [review]
LGTM assuming it works. No comment on anything outside build.sbt and project/. Some opinions:

File: project/Settings.scala (hunk 3 above)
Comment (helgridly) [reply project/Settings.scala]: I don't know what it does and I don't know what it is I want, so I'm afraid to touch it.

File: project/Settings.scala (hunk 4 above)
Comment (helgridly) [reply project/Settings.scala:48]: I don't think this is true -- it makes sense that we'd bump the Rawls version separately from the model version. Model change implies Rawls change, but not vice versa.

That said we've never used this for anything and it's all lies anyway. @dvoet any thoughts here?

**Review by dvoet — APPROVED** [review]
👍

**Pull request merged**